### PR TITLE
Update deprecation sections to reflect new policy

### PR DESCRIPTION
**Pull request by WardBrian**
#### Submission Checklist

- [x] Builds locally 
- [x] New functions marked with `` `r since("VERSION")` ``
- [x] Declare copyright holder and open-source license: see below

#### Summary

Update deprecation sections to reflect https://github.com/stan-dev/design-docs/pull/43

Related to https://github.com/stan-dev/stanc3/pull/1044

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Simons Foundation

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/reference-manual/deprecations.Rmd [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+# Deprecated Features
 
 This appendix lists currently deprecated functionality along with how
-to replace it.  These deprecated features are likely to be removed in
-the next major release.
+to replace it.
+
+Starting with Stan 2.29, minor (syntax-level) deprecations can be removed 3
+versions after release; e.g., syntax deprecated in Stan 2.20 will be removed in
+Stan 2.23 and placed in [Removed Features]. The Stan compiler can
```
Comment (bob-carpenter) [inline src/reference-manual/deprecations.Rmd:8]: same question again

File: src/reference-manual/deprecations.Rmd [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+[automatically update](https://mc-stan.org/docs/stan-users-guide/stanc-pretty-printing.html)
+many of these on the behalf of the user.
+
+Any feature which changes semantic meaning (such as the upgraded ODE solver
+interface) will not be removed until a major version change (e.g. Stan 3.0).
```
Comment (bob-carpenter) [inline src/reference-manual/deprecations.Rmd:13]: There's always a comma after "e.g.", as in `(e.g., Stan 3.0)`.

File: src/reference-manual/deprecations.Rmd [hunk 3]
```diff
@@ -123,6 +126,11 @@ target += foo_log(y, ...);
 functions or `_lpmf` for mass functions in the user-defined function.
 
 
+*Scheduled Removal*: Stan 2.32
+
+Note: Following Stan 2.32, users can stil define a function ending in `_log`,
+it will just no longer have a special meaning or support the `~` syntax.
```
Comment (bob-carpenter) [inline src/reference-manual/deprecations.Rmd:132]: Nice to see the attention to boundary conditions!

File: src/reference-manual/deprecations.Rmd [hunk 4]
```diff
@@ -123,6 +126,11 @@ target += foo_log(y, ...);
...
+
 ## `lkj_cov` distribution
 
 *Deprecated*:The distribution `lkj_cov` is deprecated.
```
Comment (bob-carpenter) [inline src/reference-manual/deprecations.Rmd:136]: There's a drop-in replacement for this.  I think it's just a lognormal prior on the scales with an LKJ prior on the correlations.  It's not a one-symbol for one-symbol replacement, though.  So probably not something that can be auto-handled by the pretty printer.

**Review by bob-carpenter — CHANGES_REQUESTED** [review]
thanks---just a few minor comments.

File: src/functions-reference/deprecated_functions.Rmd [hunk 5]
```diff
@@ -1,13 +1,54 @@
...
 
-```{r results='asis', echo=FALSE}
-if (knitr::is_html_output()) {
-  cat(' * <a href="functions-old-ode-solver.html">integrate_ode_rk45, integrate_ode_adams, integrate_ode_bdf ODE Integrators</a>\n')
-}
-```
+Starting in Stan 2.29, deprecated functions with drop in replacements (such as
+the renaming of `get_lp` or `multiply_log`) will be removed 3 versions later
+e.g., functions deprecated in Stan 2.20 will be removed in Stan 2.23 and placed
+in [Removed Functions]. The Stan compiler can
```
[the remark this answers is not in the document]
Comment (WardBrian) [reply src/functions-reference/deprecated_functions.Rmd:9]: Yes, this is an RMarkdown special case where if you want to link to a section header and the link text is the same as the header, you can omit the (). [documentation on it](https://bookdown.org/yihui/bookdown/cross-references.html)

File: src/reference-manual/deprecations.Rmd (hunk 4 above)
Comment (WardBrian) [reply src/reference-manual/deprecations.Rmd:136]: I agree, it's just barely out-of-scope in my opinion

**Review by bob-carpenter — APPROVED** [review]
Thanks.

**Comment by bob-carpenter** [comment]
It's still marked as draft, but if your intention is to merge, you should be able to do that now.

**Comment by WardBrian** [comment]
@bob-carpenter I'm going to see if any further documentation needs come up in the next couple stanc PRs related to this before merging, but if nothing changes I'll merge after those.

**Pull request merged**